### DISCUSSION TITLE
Updating Supabase table before email confirmation

### DISCUSSION THREAD

**Discussion by Dal-Tron** [supabase/supabase#3556] [General]
Current Setup:

I have a users table and a user_profile table. 

The users table is the one that comes with supabase.

What I want to do:

When a user registers, I want to also create a user profile. This just has id (user_id) and username which is a string based on the email they provide at registration. 

However, until they confirm their email... they are not technically a valid user.

Is there a RLS rule I can implement for inserting a profile that is secure? Right now I have INSERT setup to be just 'true' to allow anyone to insert...

**Comment by VictorPeralta** [comment]
You can create a database trigger on `auth.users` that does `INSERT` into your profiles table when a user is created. In the profiles table, enable RLS and don't create an `INSERT` policy, so users can't create profiles manually.
